--- a/src/plugins/clangformat/clangformatutils.cpp	2023-05-16 10:19:58.447474221 +0200
+++ b/src/plugins/clangformat/clangformatutils.cpp	2023-05-16 10:21:25.537730120 +0200
@@ -118,7 +118,7 @@
     style.ColumnLimit = 100;
     style.CommentPragmas = "^ IWYU pragma:";
     style.CompactNamespaces = false;
-    style.ConstructorInitializerAllOnOneLineOrOnePerLine = false;
+    style.PackConstructorInitializers = FormatStyle::PackConstructorInitializersStyle::PCIS_Never;
     style.ConstructorInitializerIndentWidth = 4;
     style.ContinuationIndentWidth = 4;
     style.Cpp11BracedListStyle = true;
